### PR TITLE
Add channel integration functionality for official channel detection

### PR DESCRIPTION
**Pull request by tuya-3**
## Overview
This PR implements a solution for GitHub issue #523 by adding channel integration functionality to determine if a channel is an official TUNAG channel.

## Changes
- **New file**: `server/channels/app/channel_integration.go`
  - Implements `IsOfficialChannel` function
  - Uses `INTEGRATION_ADMIN_USERNAME` environment variable for official channel detection
  - Follows Mattermost app layer architecture patterns

- **New file**: `server/channels/app/channel_integration_test.go`
  - Comprehensive test suite with 7 test cases
  - Tests normal operation, edge cases, and error conditions
  - Uses Mattermost testing patterns with Setup(t).InitBasic()

## Issues Resolved
- **[#523](https://github.com/stmninc/tunag-chat/issues/523)**: Create a common function to determine if a channel is an official channel
  - `IsOfficialChannel` function provides official channel detection capability

## Technical Details
- Direct environment variable access using `os.Getenv` following existing Mattermost patterns
- File naming follows Mattermost conventions (similar to channel.go, integration_action.go)
- Proper error handling for missing users and nil channels
- Test coverage includes unique username generation to avoid conflicts

## Testing
All tests pass successfully:
- Normal operation with valid environment variable
- Edge cases with empty environment variable
- Error handling for missing users and nil channels
- Different channel types (public, private, direct message, group message)

### CODE REVIEW

File: server/channels/app/channel_integration.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		return false, nil
+	}
+
+	creatorUser, err := a.GetUser(channel.CreatorId)
+	if err != nil {
+		return false, err
+	}
+
+	return creatorUser.Username == adminUsername, nil
+}
```
[the remark this answers is not in the document]
Comment (tuya-3) [reply server/channels/app/channel_integration.go:46]: This is not a concern for the following reasons:
- CreatorId contains system administrator username which does not change
- Username-based detection is required for SecretManager integration

File: server/channels/app/channel_integration.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	integrationAdminOnce.Do(func() {
+		integrationAdminUsername = os.Getenv("INTEGRATION_ADMIN_USERNAME")
+	})
+	return integrationAdminUsername
+}
+
+// IsOfficialChannel checks if a channel is official by comparing creator with integration admin user.
+func (a *App) IsOfficialChannel(c request.CTX, channel *model.Channel) (bool, *model.AppError) {
+	if channel == nil {
+		return false, nil
```
Comment (kondo97) [inline server/channels/app/channel_integration.go:31]: Should this be treated as an error?

Comment (tuya-3) [reply server/channels/app/channel_integration.go]: Yes, I handled it.

File: server/channels/app/channel_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+// Resets the cached integration admin username from environment variable for tests.
+func resetIntegrationAdminUsernameForTesting() {
+	integrationAdminUsername = ""
+	integrationAdminOnce = sync.Once{}
+}
+
+func TestIsOfficialChannel(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (tuya-3) [reply server/channels/app/channel_integration_test.go:26]: fixed

File: server/channels/app/channel_integration.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+	// Get cached integration admin username
+	adminUsername := getIntegrationAdminUsername()
+	if adminUsername == "" {
+		return false, model.NewAppError("IsOfficialChannel", "app.channel.config_missing", nil, "INTEGRATION_ADMIN_USERNAME not configured", http.StatusInternalServerError)
+	}
+
+	creatorUser, err := a.GetUser(channel.CreatorId)
+	if err != nil {
+		if err.StatusCode == http.StatusNotFound {
```
Comment (kondo97) [inline server/channels/app/channel_integration.go:47]: Which implementation did you reference for this?

Comment (tuya-3) [reply server/channels/app/channel_integration.go:47]:
```suggestion
		if err.Id == MissingAccountError {
```
It might be a good idea to use this standard pattern from mattermost.

File: server/channels/app/channel_integration.go (hunk 2 above)
Comment (tuya-3) [reply server/channels/app/channel_integration.go:36]: We checked the data in the channel table in the DB.
The creatorId is empty for GM and default channels (off topic etc).
